irispy/iris_tools.py

```diff
@@ -802,3 +803,33 @@ def _apply_or_undo_exposure_time_correction(sequence, correction_function):
         else:
             converted_data_list.append(cube)
     return converted_data_list
+
+def calculate_dust_mask(data_array, input_mask=None):
+    """Calculate a mask with the dust positions in a given arrayself.
+
+    Parameters
+    ----------
+    data_array : `numpy.ndarray`
+        This array contains some dust poisition that will be calculated. The array
+        must have scaled values.
+
+    Returns
+    -------
+    dust : `numpy.ndarray` of `bool`
+        This array has the same shape than data_array and contains the dust positions
+        when the value is True.
+
+    """
+    # Creating a mask with the same shape than the inputed data array.
+    mask = np.zeros_like(data_array, dtype=bool)
+    # Set the pixel value to True is the pixel is recognized as a dust pixel.
+    mask[(data_array < 0.5) & (data_array > -200)] = True
+    # Extending the mask to avoid the neighbours pixel influenced by the dust pixels.
+    struct = np.array([np.zeros((3, 3)), np.ones((3, 3)), np.zeros((3, 3))], dtype=bool)
```

Comment:
To use `ndimage.binary_dilation`, I need to input a `structure` kwarg. This kwarg is like a template for the dilation. Here, our dust pixel is the `struct[1,1,1]` pixel. By setting `np.ones((3,3))` for `struct[1]`, we say that all neighbours (diagonal included) of our dust pixel must be considerate as a dust pixel. About `stuct[0]` and `struct[2]`, I added them as Tiago suggested me to avoid the previous loop I was using. Their translation is " In the previous and the next frame do : Nothing" due to the `np.zeros((3,3)).

The `structure` kwarg is only dependent on what we need to do as I said before. The most important thing is if we not loop, we must have a 3D kwarg.

Is it clearer now ?

Reply:
OK, this makes more sense now :)